compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -823,6 +825,22 @@ export function printFunctionType(type: ReactFunctionType): string {
   }
 }
 
+export const EMIT_FREEZE_GLOBAL_GATING = '__DEV__';
+export type ImportedUids = {
+  useMemoCacheIdentifier: string;
+  gating: string | null;
+  lowerContextAccess: string | null;
+  enableEmitInstrumentForget: {
+    fn: string;
+    gating: string | null;
+    globalGating: string | null;
+  } | null;
+  enableEmitFreeze: string | null;
+  enableEmitHookGuards: string | null;
+  enableChangeDetectionForDebugging: string | null;
+  enableFire: string | null;
+};
```

Comment:
It's pretty gross that we're producing a sidemap of external functions and duplicating each external function config.

One alternative is to instead keep a `Map<ExternalFunction, string>` and expose a `Environment.getLocalUid(fn: ExternalFunction): string`. 

We can also annotate each `ExternalFunction` definition with a `localVariableName: string | null` property, then populate when parsing EnvironmentConfigs. This would feel the cleanest types-wise, but I don't love the idea of custom, per-program EnvironmentConfigs